pkg/k8s/watchers/watcher.go

```diff
@@ -324,6 +324,9 @@ func (k *K8sWatcher) resourceGroups() []string {
 	if k8s.SupportsEndpointSlice() {
 		k8sGroups = append(k8sGroups, K8sAPIGroupEndpointSliceV1Beta1Discovery)
 	}
+	if k8s.SupportsEndpointSliceV1() {
+		k8sGroups = append(k8sGroups, K8sAPIGroupEndpointSliceV1Discovery)
+	}
```

Comment:
this needs to be made into an else statement. The agent does not listen for both versions at the same time, it's either one or the other but not both.